wagtail/wagtailcore/blocks/list_block.py

```diff
@@ -113,19 +129,27 @@ def value_omitted_from_data(self, data, files, prefix):
 
     def clean(self, value):
         result = []
-        errors = []
-        for child_val in value:
+        errors = {}
+        for i, child_val in enumerate(value):
             try:
                 result.append(self.child_block.clean(child_val))
             except ValidationError as e:
-                errors.append(ErrorList([e]))
-            else:
-                errors.append(None)
+                errors[i] = ErrorList([e])
 
-        if any(errors):
+        max_length = self.list_options['max_length']
+        if max_length and max_length < len(value):
+            raise ListBlockValidationError(non_block_errors=ErrorList([
+                ValidationError(_("Maximum of %s is reached" % max_length))]))
```

Comment:
This error message feels very impersonal, and not as clear as it could be. Perhaps something like "The maximum number of items is %s"